enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/hooks/use-param-rerender-key.ts

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from "react";
+import { P, match } from "ts-pattern";
+
+import { sortObject } from "metabase-lib/v1/utils";
+
+import type { SdkIframeEmbedSettings } from "../../embedding_iframe_sdk/types/embed";
+
+/**
+ * Forces a re-render when the initial parameter changes.
+ */
+export const useParamRerenderKey = (settings: SdkIframeEmbedSettings) =>
+  useMemo(
+    () =>
+      match(settings)
+        .with(
+          { dashboardId: P.nonNullable },
+          (settings) =>
+            `dashboard-${stableStringify(settings.initialParameters)}`,
+        )
+        .with(
+          { questionId: P.nonNullable },
+          (settings) =>
+            `question-${stableStringify(settings.initialSqlParameters)}`,
+        )
+        .otherwise(() => null),
+    [settings],
+  );
+
+// Stringify with sorted keys to ensure stable orders.
+const stableStringify = <T>(obj: T): string => JSON.stringify(sortObject(obj));
```

Comment:
TIL: about `sortObject` how did you find this 😄

Reply:
I just search for `sortObject` and found it, lucky!